subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -0,0 +1,153 @@
+        super(deferred_deletion, self).run_once()
+
+        self._start_idle_container()
+
+        # cd into its directory. This prevents devicemapper from reaping it.
+        os.chdir(self._container_mountpoint)
+
+        # Determine baseline count of deferred deletions. We expect 0,
+        # but let's not panic if it isn't.
+        defer_count_initial = self._defer_count()
+        if defer_count_initial != 0:
+            self.logwarning("Deferred-Delete count from 'docker info'"
+                            " is %d (I expected 0). This may mean your"
+                            " system has still-undeleted containers.",
+                            defer_count_initial)
+        self.stuff['defer_count_initial'] = defer_count_initial
+
+        # Tell the container to exit
+        self._stop_idle_container()
+
+    def postprocess(self):
+        super(deferred_deletion, self).postprocess()
+
+        defer_count_initial = self.stuff['defer_count_initial']
+        self.failif_ne(self._defer_count(),
+                       defer_count_initial + 1,
+                       "Deferred-Delete count while cd'ed into container")
+
+        # cd out of the container filesystem. devmapper will notice and
+        # will clean up, but it may take a while. Allow up to 30 seconds.
+        os.chdir('/')
+        defer_count_back = lambda: self._defer_count() == defer_count_initial
+        self.failif(not utils.wait_for(defer_count_back, timeout=30),
+                    "Timed out waiting for Deferred-Delete count (%d) to"
+                    " come back down to %d after cd'ing out of container"
+                    % (self._defer_count(), defer_count_initial))
+
+    def _start_idle_container(self):
+        """
+        Start a container. We only need it briefly, until we (the test,
+        running in host space) can cd into its root filesystem. Container
+        will spin idly until a semaphore file is removed.
+        """
+        self.stuff['dc'] = dc = DockerContainers(self)
+        c_name = dc.get_unique_name()
+
+        fin = DockerImage.full_name_from_defaults(self.config)
+        subargs = ['--rm', '--name=' + c_name, fin,
+                   'bash -c "echo READY;touch /DELETE-ME;'
+                   ' while [ -e /DELETE-ME ]; do sleep 0.1; done"']
+        dkrcmd = AsyncDockerCmd(self, 'run', subargs)
+        dkrcmd.execute()
+        dkrcmd.wait_for_ready(c_name)
+        self.stuff['container_name'] = c_name
+
+    def _stop_idle_container(self):
+        """
+        Stop our idle container, by removing its semaphore file.
+        Wait until container is truly gone, as reported by docker ps -a
+        """
+        os.unlink('DELETE-ME')
+
+        dc = self.stuff['dc']
+        c_name = self.stuff['container_name']
+        c_gone = lambda: dc.list_containers_with_name(c_name) == []
+        self.failif(not utils.wait_for(c_gone, timeout=5),
+                    "Timed out waiting for container to exit")
```

Comment:
How about attempting to kill this guy if it exists, during ``cleanup()``?

(otherwise, letting ``garbage_check`` grab it is probably okay)